src/main/java/ru/yandex/qatools/fsm/impl/YatomataImpl.java

```diff
@@ -15,7 +15,7 @@
 /**
  * @author Ilya Sadykov
  */
-public class YatomataImpl<T> implements Yatomata<T> {
+class YatomataImpl<T> implements Yatomata<T> {
```

Comment:
@vania-pooh we probably have to add the ability to instantiate the FSM with the reference FSM instance from the FSMBuilder. Impl classes usually have to be non-public. 
Can you send a separate PR for that?